sentry-ruby/lib/sentry/transport/http_transport.rb

```diff
@@ -13,6 +18,10 @@ def initialize(*args)
     end
 
     def send_data(data)
+      if should_compress?(data)
+        data = Base64.strict_encode64(Zlib.deflate(data))
```

Comment:
This uses our legacy compression format, which uses deflate + base64, not gzip. While we technically support it on the envelope endpoint, it has a few drawbacks. Could you instead set the `Content-Encoding` encoding header and apply actual gzip compression?

If gzip is not an option, you can also set the content encoding to `deflate` and then do a full `zlib` compression (including the zlib structure header). 

See [Request Compression](https://develop.sentry.dev/sdk/overview/#request-compression) and [`Content-Encoding`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding) for more information.